lib/parser/char-block.h

```diff
@@ -74,6 +74,13 @@ class CharBlock {
   bool operator>=(const CharBlock &that) const { return Compare(that) >= 0; }
   bool operator>(const CharBlock &that) const { return Compare(that) > 0; }
 
+  bool operator<(const char *that) const { return Compare(that) < 0; }
+  bool operator<=(const char *that) const { return Compare(that) <= 0; }
+  bool operator==(const char *that) const { return Compare(that) == 0; }
+  bool operator!=(const char *that) const { return Compare(that) != 0; }
+  bool operator>=(const char *that) const { return Compare(that) >= 0; }
+  bool operator>(const char *that) const { return Compare(that) > 0; }
```

Comment:
These are useful functions but using operators makes me uneasy. Especially equality, which should be symmetric.

Reply:
What if I make them non-member friend overloads?

Reply:
My concern is just that with these operators you can have `x == y` but you can't rewrite it as `y == x`. Similarly, `x > y` is not equivalent to `y < x`.

But the alternatives give you compilation errors (as opposed to getting the wrong answer), so maybe it isn't a big deal.

Reply:
I should be more explicit.  Please see the current source.  The point of turning (some) of those member functions into friends was to enable the symmetric overloads.

Reply:
I get it now. Yes, that addresses my concerns.